tools/write_changelog_entry.py

```diff
@@ -0,0 +1,68 @@
+import os
+import re
+import glob
+
+
+def inc(obj, key):
+    """Increment a dictionary[key] value."""
+    try:
+        obj[key] += 1
+    except KeyError:
+        obj[key] = 1
+
+
+# Load PR number and comment
+try:
+    pr = os.environ['PR_NUMBER']
+    body = os.environ['PR_BODY']
+except KeyError:
+    raise ValueError('PR_NUMBER and PR_BODY env vars must be set.')
+assert str(int(pr)) == pr
+body = body.replace('\\r', '').replace('\\n', '\n')
+
+# Get changelog directory
+# TODO: get directory from pyproject.toml
+path = os.path.join(os.getenv('CHANGELOG_DIR', 'changelog'))
+if not os.path.exists(path):
+    raise ValueError(f"CHANGELOG_DIR path '{path}' must exist.")
+
+# Search for changelog entries
+#     markdown heading of "Changelog [category]"
+#     followed by a message in the next ``` code block.
+body = re.sub("(<!--.*?-->)", "", body, flags=re.DOTALL)
+regex = r"[#]+[ ]*Changelog[ ]*\[([a-zA-Z]+)\]*[^`]*```([^`]*)```"
```

Comment:
If we don't find anything in body of the PR can we terminate the action from here?

Reply:
Currently, if you remove the a Changelog from the main PR comment it deletes that Changelog from the PR branch. If we set it to exit here it would make deleting changelogs inconsistent, I.e: if you have two Changelog entries (feature and bugfix) and remove the feature Changelog from the comment it will be deleted in the PR branch, however, if you then remove the bugfix Changelog (so no regex match) it won't be deleted from the PR branch.

Reply:
Hmmm, that does make sense, but I don't think we should be deleting files people are adding manually either. If we can't delete entries from the PR description because of that, then I think that's a price we would have to pay. Unless we can come up with some other idea.